src/fromager/commands/download_sequence.py

```diff
@@ -0,0 +1,30 @@
+import logging
+import json
+
+import click
+from packaging.requirements import Requirement
+
+from .. import sources
+
+logger = logging.getLogger(__name__)
+
+
+@click.command()
+@click.argument('build_order_file')
+@click.argument('sdist_server_url')
+@click.pass_obj
+def download_sequence(wkctx, build_order_file, sdist_server_url):
+    """Download a sequence of source distributions in order.
+
+    BUILD_ORDER_FILE is the build-order.json files to build
+
+    SDIST_SERVER_URL is the URL for a PyPI-compatible package index hosting sdists
+
+    Performs the equivalent of the 'step download-source-archive' command for each item in
+    the build order file.
+
+    """
+    with open(build_order_file, 'r') as f:
+        for entry in json.load(f):
```

Comment:
The loop is nice and straightforward. It may be fairly slow, though. Did you give any thought to parallelizing it? `futures.ThreadPoolExecutor()` is an easy way to do that. There's an example in https://gitlab.com/redhat/rhel-ai/wheels/pipeline-tool/-/blob/main/src/pipeline_tool/jobs.py?ref_type=heads#L58

Let's take this as-is, and consider whether we ought to make that enhancement in another PR.